test/test_puma_server_ssl.rb

```diff
@@ -33,7 +33,17 @@ def ssl_error(server, peeraddr, peercert, error)
 class TestPumaServerSSL < Minitest::Test
 
   def setup
-    return if DISABLE_SSL
+    @http = nil
+    @server = nil
+  end
+
+  def teardown
+    @http.finish if !@http.nil? && @http.started?
```

Comment:
style nitpick but I hate unnecessary predicates. I think `if @http` is sufficient here, as is `unless @server` in the next line.

Reply:
@nateberkopec

I just pushed this change, and it's breaking things.

Although I can't recall what/why, there are times when connections aren't 'made' due to testing for restricted protocol versions.  Are you ok with returning these to the original code in the PR?

Reply:
Yup! if the predicate is necessary, then it's necessary. just thought it wasn't.